FreeRTOS-Plus/Test/CBMC/proofs/prvProcessEthernetPacket/prvProcessEthernetPacket_harness.c

```diff
@@ -0,0 +1,84 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_UDP_IP.h"
+#include "FreeRTOS_DHCP.h"
+#include "NetworkInterface.h"
+#include "NetworkBufferManagement.h"
+#include "FreeRTOS_ARP.h"
+
+#include "cbmc.h"
+
+
+/****************************************************************
+ * Signature of function under test
+ ****************************************************************/
+
+void prvProcessEthernetPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer );
```

Comment:
Is there any sanity checking we should be doing for ProcessEthernetPacket and RefreshCacheEntry?  At least asserting pointers are not null?  Are there any assumptions used in the memory safety proof of these functions that we should be asserting?

Reply:
Added sanity checks.